pay-api/src/pay_api/services/eft_short_names.py

```diff
@@ -602,36 +603,55 @@ def get_search_query(cls, search_criteria: EFTShortnamesSearch, is_count: bool =
             ))
 
             # Short name link filters
-            query = query.filter_conditionally(search_criteria.id, EFTShortnameModel.id)
-            query = query.filter_conditionally(search_criteria.account_id,
-                                               EFTShortnameLinksModel.auth_account_id,
-                                               is_like=True)
+            subquery = subquery.filter_conditionally(search_criteria.account_id,
+                                                     EFTShortnameLinksModel.auth_account_id,
+                                                     is_like=True)
+
             # Payment account filters
-            query = query.filter_conditionally(search_criteria.account_name, PaymentAccountModel.name, is_like=True)
-            query = query.filter_conditionally(search_criteria.account_branch, PaymentAccountModel.branch_name,
-                                               is_like=True)
+            subquery = subquery.filter_conditionally(
+                search_criteria.account_name, PaymentAccountModel.name, is_like=True)
+            subquery = subquery.filter_conditionally(search_criteria.account_branch, PaymentAccountModel.branch_name,
+                                                     is_like=True)
 
             # Statement summary filters
-            query = query.filter_conditionally(search_criteria.statement_id,
-                                               statement_summary_query.c.latest_statement_id)
+            subquery = subquery.filter_conditionally(search_criteria.statement_id,
+                                                     statement_summary_query.c.latest_statement_id)
             if search_criteria.amount_owing == 0:
-                query = query.filter(or_(statement_summary_query.c.total_owing == 0,
-                                         statement_summary_query.c.total_owing.is_(None)))
+                subquery = subquery.filter(or_(statement_summary_query.c.total_owing == 0,
+                                               statement_summary_query.c.total_owing.is_(None)))
             else:
-                query = query.filter_conditionally(search_criteria.amount_owing, statement_summary_query.c.total_owing)
+                subquery = subquery.filter_conditionally(
+                    search_criteria.amount_owing, statement_summary_query.c.total_owing)
 
-        query = cls.get_link_state_filters(search_criteria, query)
-        query = query.filter(or_(CfsAccountModel.payment_method == PaymentMethod.EFT.value,
-                                 CfsAccountModel.id.is_(None)))
-        query = query.filter(
+        subquery = cls.get_link_state_filters(search_criteria, subquery)
```

Comment:
these aren't quite right, they should be main queries that filter out the data not just the subquery